pkg/app/app_factory.go

```diff
@@ -29,13 +29,12 @@ type CRDAppFactory struct {
 }
 
 // NewCRDApp creates a CRDApp injecting necessary dependencies.
-func (f *CRDAppFactory) NewCRDApp(app *kcv1alpha1.App, log logr.Logger) *CRDApp {
+func (f *CRDAppFactory) NewCRDApp(app *kcv1alpha1.App, log logr.Logger, deployFactory deploy.Factory) *CRDApp {
```

Comment:
instead of passing down deployFactory into NewCRDApp, let's make CRDAppFactory constructor accept it. this way whoever calls NewCRDApp does not need to know anything about deployFactory.

Reply:
By constructor do you mean just add it as another public field on the struct? There is no constructor right now for CRDAppFactory. We could make one, and then keep it as a private field?

Reply:
@cppforlife